### DISCUSSION TITLE
Feedback about AFFiNE UI/UX and self hosting

### DISCUSSION THREAD

**Discussion by ZoneTwelve** [toeverything/AFFiNE#4027] [General]
## Introduction

> **Warning**
> This issue comes from **Mar 4**
> Because of some network issues, the post has come late.

a user uses AFFiNE, I got a few stuff think will make AFFiNE better.

## User Experience

- Not easy to understand
  - Problem:
    - Upload pictures need to drag the button into the block area or use the slash menu.
    - Until someone tell me the whiteboard feature is in edgeless mode, I didn't know the whiteboard exist.
  - Solution / Suggestion:
    - Quick tutorial before using it
    - Just allow the user to drag a file/image to upload it 
- Local self-hosting
  - Problem:
    - Allow user login with local auth system
    - an easy way to deploy the latest application (hear from other users)
  - Solution:
    - Use local database
    - Create a deployment tool like docker-compose 
      - Ex. [AFFiNE Composer](https://github.com/ZoneTwelve/AFFiNE-Composer) still have a few bugs in there.

**Comment by houjoe0829** [comment]
Thank you for you feedback 
1. Dragging a picture to add to the editor is in the plan  https://github.com/toeverything/blocksuite/issues/1447
2. Regarding the differentiation between the whiteboard and the page, we will pay attention to it in subsequent design optimizations, as there is indeed a lot of room for improvement.

As for the Self-host, @tzhangchi please take a look at that.

**Comment by tzhangchi** [comment]
Hi, ZoneTwelve! It's great to meet you. I just saw that you've mentioned a problem about `Allow user login with local auth system` in the AFFiNE repository. We have created  a discussion about it https://github.com/toeverything/AFFiNE/discussions/1142

**Comment by ZoneTwelve** [comment]
Thank you for taking the time to respond to my feedback. Your insights and suggestions have been incredibly helpful and I appreciate the effort you have put into addressing my concerns. I will certainly keep an eye out for any important issues and report them as needed. Once again, thank you for your time and attention.
